sql/common/src/main/scala/org/apache/spark/sql/sedona_sql/strategy/join/JoinedGeometry.scala

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.spark.sql.sedona_sql.strategy.join
+
+import org.apache.sedona.common.utils.GeomUtils
+import org.geotools.coverage.grid.GridCoverage2D
```

Comment:
I suggest that we move geotools related imports and their functions such as `org.geotools` and `org.opengis` to a separate class called `JoinedGeometryGeoTools` or `JoinedGeometryRaster`.

Packaging Geotools-wrapper might be problematic on some cloud environments. Separating geotools imports will prevent the core Sedona vector join logic from getting `Geotools Class Not Found` issue. Otherwise, all Sedona vector joins will fail.

This is related to this PR: https://github.com/apache/sedona/pull/966